yarn-project/prover-node/src/bond/token-contract.ts

```diff
@@ -50,6 +50,29 @@ export class TokenContract {
     }
   }
 
+  /**
+   * Checks the sender address has enough balance.
+   * If it doesn't, it tries calling a `mint` method, available on testing environments.
+   * If it can't, it throws an error.
+   * @param amount - The balance to ensure.
+   */
+  public async ensureBalance(amount: bigint) {
+    const balance = await this.getBalance();
+    if (balance < amount) {
+      this.logger.verbose(`Balance ${balance} is below required ${amount}. Attempting mint.`);
+      const testToken = getContract({ address: this.token.address, abi: TestERC20Abi, client: this.client });
+      try {
+        await testToken.simulate.mint([this.getSenderAddress().toString(), amount - balance]);
```

Comment:
What is the function of the simulate call here, given it has no return value?

Reply:
Checking whether the tx will succeed. If it'd revert, eg because the `mint` function does not exist in the contract, then `simulate` would throw.